tariff/proxy.go

```diff
@@ -0,0 +1,234 @@
+package tariff
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"sync"
+	"sync/atomic"
+	"time"
+	"unsafe"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/util"
+)
+
+// CachingTariffProxy wraps a tariff provider to add caching functionality
+type CachingTariffProxy struct {
```

Comment:
```suggestion
type Proxy struct {
```

try to match file name where possible. `Tariff` stutters since we're already in the tariff package- what else should be here?